src/main/scala/com/dwolla/sbt/s3/PublishToS3.scala

```diff
@@ -28,12 +28,14 @@ object PublishToS3 extends AutoPlugin {
   )
 
   lazy val tasks = Seq(
-    s3PublishSnapshot := !(git.gitCurrentBranch.value == "master" && !git.gitUncommittedChanges.value),
+    s3PublishSnapshot := !(
+      (git.gitCurrentBranch.value == "master" || git.gitHeadCommit.value.contains(git.runner.value("rev-parse master")(baseDirectory.value, streams.value.log)))
```

Comment:
I know it was already here once, but I don't love that `master` branch is hard-coded here. I suggest we introduce a new setting called something like `publishBranch` and give it a default value of `"master"`. Projects would then be free to override the branch name as they see fit.

e.g., in `S3PublisherPluginKeys.scala`, to add the new setting key:
```scala
  lazy val publishBranch = settingKey[String]("branch name from which canonical artifacts should be published")
```

and then in `PublishToS3.scala`, add this to the `defaultValues` list:

```scala
    publishBranch := "master",
```

Then the `"master"` string literals would be replaced with `publishBranch.value`. If a project using this plugin overrides the value in its `build.sbt`, the replacement value will be used in our checks.